franklin-crypto/src/circuit/rescue.rs

```diff
@@ -0,0 +1,271 @@
+extern crate num_bigint;
+use self::num_bigint::{BigUint,ToBigUint};
+use std::str::FromStr;
+use super::uint32::UInt32;
+use super::multieq::MultiEq;
+use super::boolean::Boolean;
+use bellman::{ConstraintSystem, SynthesisError};
+use bellman::pairing::Engine;
+use circuit::Assignment;
+use bellman::pairing::ff::{
+    Field,
+    SqrtField,
+    PrimeField,
+    PrimeFieldRepr,
+    BitIterator
+};
+use bellman::{
+    LinearCombination,
+    Variable
+};
+use super::num::AllocatedNum;
+
+
+pub fn generate_powers<CS,E>(
+    mut cs: CS,
+    x: &AllocatedNum<E>,
+    alpha: &E::Fr
+) -> Result<AllocatedNum<E>, SynthesisError>
+    where CS: ConstraintSystem<E>,E:Engine
+{
+        assert_ne!(*alpha,E::Fr::zero());
+        assert_ne!(*alpha,E::Fr::one());
+        let alpha_bits_be = {
+            let mut tmp = Vec::with_capacity(E::Fr::NUM_BITS as usize);
+            let mut found_one = false;
+            for b in BitIterator::new(alpha.into_repr()) {
+                found_one |= b;
+                if !found_one {
+                    continue;
+                }
+                tmp.push(b);
+            }
+            tmp
+        };
+        let squares={//define square, square of square, etc variables
```

Comment:
it's ok for now to compute squares each time, but for production code it would be a waste of resources. 

to cache squeres for different alphas consider
https://docs.rs/lazy_static/1.4.0/lazy_static/